feat(manager/custom): allow packageName instead of depName (#28834)
renovatebot/renovate: lib/modules/manager/custom/regex/readme.md

Configuration-wise, it works like this:

- You must capture the `currentValue` of the dependency in a named capture group
- You must have either a `depName` capture group or a `depNameTemplate` config field, or `packageName` or `packageNameTemplate`

> - You can optionally have a `packageName` capture group or a `packageNameTemplate` if it differs from `depName`

Reviewer: I think this needs to be removed?

> - You can optionally have a `packageName` capture group or a `packageNameTemplate` if it differs from `depName`

Author: ```suggestion
```

Configuration-wise, it works like this:

- You must capture the `currentValue` of the dependency in a named capture group

> - You must have either a `depName` capture group or a `depNameTemplate` config field, or `packageName` or `packageNameTemplate`

Reviewer 2: ```suggestion
- You must have either a `depName` or `packageName` capture group, or use on of the respective template fields (  `depNameTemplate` and `packageNameTemplate` )
```

Instead you enhance your `Dockerfile` like this:

```Dockerfile
# renovate: datasource=github-tags depName=node packageName=nodejs/node versioning=node
ENV NODE_VERSION=20.10.0
# renovate: datasource=github-releases depName=composer packageName=composer/composer
ENV COMPOSER_VERSION=1.9.3
# renovate: datasource=docker packageName=docker versioning=docker
ENV DOCKER_VERSION=19.03.1

> # renovate: datasource=npm packageName=yarn

Reviewer 2: ```suggestion
# renovate: datasource=npm depName=YARN packageName=yarn
```
Example of using both.

Author: There's already two examples earlier so I preferred to have this one without

Instead you enhance your `Dockerfile` like this:

```Dockerfile
ARG IMAGE=node:12@sha256:6e5264cd4cfaefd7174b2bc10c7f9a1c2b99d98d127fc57a802d264da9fb43bd

> FROM ${IMAGE}

Author: This type of pattern is now handled natively by the `dockerfile` manager so is not a good example for regex manager
